### PR TITLE
[DOC] change links in documentation to refer to same version

### PR DESCRIPTION
**Pull request by yarnabrina**
In most places in `docs/source`, links are hardcoded to either `latest` (corresponding to `main` branch) or to `stable` (most recent tag). But quite often users will probably prefer to see the documentation for the same version they are browsing, assuming because they are using that version at the momemt. This PR makes these changes, following https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-doc-role.

### CODE REVIEW

File: docs/source/developer_guide.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ New developers should:
 * install a development version of ``sktime``, see :ref:`installation`
 * set up CI tests locally and ensure they know how to check them remotely, see :ref:`continuous_integration`
 * get familiar with the git workflow (:ref:`git_workflow`) and coding standards (:ref:`coding_standards`)
-* feel free, at any point in time, to post questions on Discord, or ask core developers for help (see here for a `list of core developers <https://www.sktime.net/en/stable/about/team.html>`_)
+* feel free, at any point in time, to post questions on Discord, or ask core developers for help (see here for a `list of core developers :doc:</about/team>`)
```
Comment (fkiraly) [inline docs/source/developer_guide.rst:21]: same about this one

File: docs/source/get_involved/governance.rst [hunk 2]
```diff
@@ -196,7 +196,7 @@ to the continued development of the project through ongoing engagement
 with the community.
 
 Current core developers are listed in the `core-developers
```
Comment (fkiraly) [inline docs/source/get_involved/governance.rst:198]: same

**Review by fkiraly — CHANGES_REQUESTED** [review]
I agree with the software documentation, but roles/governance I would leave pointing to the current state, reasoning as above.

File: docs/source/get_involved/governance.rst [hunk 3]
```diff
@@ -740,7 +740,7 @@ References
 Our governance model is inspired by various existing governance
 structures. In particular, we'd like to acknowledge:
 
-* scikit-learn's `governance model <https://www.sktime.net/en/latest/governance.html>`__
+* scikit-learn's `governance model <https://scikit-learn.org/stable/governance.html>`__
```
Comment (yarnabrina) [inline docs/source/get_involved/governance.rst:743]: Not sure why `sktime` link was used here. Changed to that of `scikit-learn`, please confirm.

**Review by yarnabrina — COMMENTED** [review]
@fkiraly reverted the specified lines. Requesting re-review, specifically the below one.

File: docs/source/get_involved/governance.rst (hunk 3 above)
Comment (fkiraly) [reply docs/source/get_involved/governance.rst:743]: confirmed, yes, of course

**Review by fkiraly — APPROVED** [review]
Looks good, only the docs build is failing - I think it's not related?

**Comment by fkiraly** [comment]
updated from main, docs have built now

**Pull request merged**